### PR TITLE
Increased logging to console and introduced file logging

### PR DESCRIPTION
**Pull request by andreachild**
Increased logging to console and introduced file logging to facilitate troubleshooting.

File logging:
* uses pino and pino-pretty logging libraries
* log level is controlled via the `--log-verbose option` which will output more information to the file (default log level is INFO)
* queries and node/edge data are logged at DEBUG level and not logged by default

Console output:
* added more messages and spinners to indicate what the utility is doing and its progress
* yellow text is used to highlight some information for the user

Other:
* fixed sample TODO schema which had extra data that causes it to fail when used as an import schema
* deleted unused test case input graphql schema files

### CODE REVIEW

File: src/pipelineResources.js [hunk 1]
```diff
@@ -633,145 +713,174 @@ async function removeAWSpipelineResources(resources, quietI) {
...
         };
         let command = new DetachRolePolicyCommand(input);        
         await iamClient.send(command);
-        if (!quiet) spinner.succeed('Detached policy: ' + yellow(resources.LambdaExecutionPolicy1) + " from role: " + yellow(resources.LambdaExecutionRole));
+        succeedSpinner('Detached policy: ' + yellow(resources.LambdaExecutionPolicy1) + " from role: " + yellow(resources.LambdaExecutionRole), {logLevel: 'debug'});
+        loggerInfo('Detached first IAM policy from role');
     } catch (error) {
-        if (!quiet) spinner.fail('Detach policy failed: ' + error);
+        let message = 'Detach first policy failed';
+        if (!quiet) spinner.fail(message);
```
Comment (kmcginnes) [inline src/pipelineResources.js:761]: Should the `spinner.fail()` call also have an abstracted helper function like `spinnerSucceed()` and `spinnerWarn()`?

File: src/logger.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+let logFileDestination;
+
+/**
+ * Initialize the standard out and file loggers.
+ * @param directory the directory in which to create the log file
+ * @param quiet true if the standard output should be minimalized to errors only
+ * @param logLevel the file log level
+ */
+function loggerInit(directory, quiet = false, logLevel = 'info') {
+    logFileDestination = directory + '/log_' + (new Date()).toISOString() + '.txt';
```
[the remark this answers is not in the document]
Comment (andreachild) [reply src/logger.js]: Changed to use path

File: src/lambdaZip.js [hunk 3]
```diff
@@ -1,5 +1,8 @@
 import fs from 'fs';
 import archiver from 'archiver';
+import { loggerError } from "./logger.js";
+
+let msg = '';
```
[the remark this answers is not in the document]
Comment (andreachild) [reply src/lambdaZip.js]: Removed

File: src/pipelineResources.js [hunk 4]
```diff
@@ -80,14 +81,50 @@ let ZIP = null;
...
+ * Stop the spinner with a success message.
+ *
+ * To also log the message to file, set logOptions.logLevel to 'info' or 'debug'. Do not log messages with sensitive data to file at info level.
+ *
+ * @param text the text to display on the spinner to indicate success
+ * @param logOptions optional logging options which determine if the text is also logged to file and what level it should be logged at
+ */
+function succeedSpinner(text, logOptions = {logLevel: ''}) {
+    if (!quiet) {
+        spinner.succeed(text);
```
[the remark this answers is not in the document]
Comment (andreachild) [reply src/pipelineResources.js:114]: Added check that spinner has been started before succeeding.

File: src/pipelineResources.js [hunk 5]
```diff
@@ -80,14 +81,50 @@ let ZIP = null;
...
+    if (logOptions.logLevel === 'info') {
+        loggerInfo(text);
+    } else if (logOptions.logLevel === 'debug') {
+        loggerDebug(text);
+    }
+}
+
+function warnSpinner(text) {
+    if (!quiet) {
+        spinner.warn(text);
```
[the remark this answers is not in the document]
Comment (andreachild) [reply src/pipelineResources.js:128]: Added check that spinner has been started before warning.

File: src/pipelineResources.js [hunk 6]
```diff
@@ -358,15 +430,17 @@ async function createLambdaFunction() {
     const lambdaClient = new LambdaClient({region: REGION});
     const data = await lambdaClient.send(new LambdaCreateFunctionCommand(params));
     LAMBDA_ARN = data.FunctionArn;
-    storeResource({LambdaFunction: NAME +'LambdaFunction'});
-    if (!quiet) spinner.succeed('Lambda Name: ' + yellow(NAME +'LambdaFunction') + ' ARN: ' + yellow(LAMBDA_ARN));
+    storeResource({LambdaFunction: lambdaName});
+    succeedSpinner('Lambda Name: ' + yellow(NAME +'LambdaFunction') + ' ARN: ' + yellow(LAMBDA_ARN), {logLevel: 'debug'});
```
[the remark this answers is not in the document]
Comment (andreachild) [reply src/pipelineResources.js]: Reused `lambdaName`

File: src/pipelineResources.js (hunk 1 above)
Comment (andreachild) [reply src/pipelineResources.js]: Extracted function `failSpinner`

**Review by kmcginnes — APPROVED** [review]
LGTM

**Pull request merged**